server/src/main/java/com/thoughtworks/go/server/newsecurity/providers/AccessTokenBasedPluginAuthenticationProvider.java

```diff
@@ -67,16 +71,21 @@ protected String getUsername(AuthenticationToken<AccessTokenCredential> authenti
 
     @Override
     protected boolean doesPluginSupportAuthentication(String pluginId) {
-        return store.doesPluginSupportTokenBasedAuthentication(pluginId);
+        return true;
     }
 
     @Override
     protected AuthenticationResponse authenticateWithExtension(String pluginId,
                                                                AccessTokenCredential credentials,
                                                                SecurityAuthConfig authConfig,
                                                                List<PluginRoleConfig> pluginRoleConfigs) {
-        return authorizationExtension.authenticateUser(pluginId, credentials.getAccessToken().getUsername(),
-                singletonList(authConfig), pluginRoleConfigs);
+        if (store.doesPluginSupportGetUserRolesCall(pluginId)) {
+            return authorizationExtension.getUserRoles(pluginId, credentials.getAccessToken().getUsername(), singletonList(authConfig), pluginRoleConfigs);
+        }
+
+        com.thoughtworks.go.domain.User fetched = userService.findUserByName(credentials.getAccessToken().getUsername());
```

Comment:
Should also check for a user account is disabled by gocd admin or not?

Reply:
@bdpiparva -- agree. We should authenticate the current user before proceeding.

Though, we don't have a way to validate the current user.

The existing authenticate authorization endpoint requires us to send, username and credentials for authentication, whereas we do not have this information for the access_token API call. also, we really are not authenticating the user, but asking for the validity (still exists).

Hence, We've decided to introduce another extension endpoint saying `does_user_exists` which will return 200 if the provided user exists in the auth_config.

I'll be sending another PR to make these changes instead of piling up more changes on the existing PR.